mpi-serial build broken

undefined reference to `MPI_Type_create_hvector'
I thought that there was a workaround in the code for this, did it get removed?
http://my.cdash.org/buildSummary.php?buildid=1157043

You should just define the new function in your MPI serial library. It is
exactly the same as MPI_Type_create_hvector except for the name.

On Mon, Mar 20, 2017 at 7:08 AM, jedwards4b <notifications@github.com>
wrote:

> undefined reference to `MPI_Type_create_hvector'
> I thought that there was a workaround in the code for this, did it get
> removed?
> http://my.cdash.org/buildSummary.php?buildid=1157043
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NCAR/ParallelIO/issues/925>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEUr3BA6l2w3xeC2sJWfibHaX5b6e2ePks5rnnpHgaJpZM4MiZx7>
> .
>

Sorry, I meant its the same as MPI_Type_hvector.

On Mon, Mar 20, 2017 at 8:17 AM, Ed Hartnett <edwardjameshartnett@gmail.com>
wrote:

> You should just define the new function in your MPI serial library. It is
> exactly the same as MPI_Type_create_hvector except for the name.
>
> On Mon, Mar 20, 2017 at 7:08 AM, jedwards4b <notifications@github.com>
> wrote:
>
>> undefined reference to `MPI_Type_create_hvector'
>> I thought that there was a workaround in the code for this, did it get
>> removed?
>> http://my.cdash.org/buildSummary.php?buildid=1157043
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/NCAR/ParallelIO/issues/925>, or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AEUr3BA6l2w3xeC2sJWfibHaX5b6e2ePks5rnnpHgaJpZM4MiZx7>
>> .
>>
>
>

--- a/src/clib/pio_rearrange.c
+++ b/src/clib/pio_rearrange.c
@@ -880,9 +880,15 @@ int rearrange_comp2io(iosystem_desc_t *ios, io_desc_t *iodesc, void *sbuf,
 
                     /*  The stride here is the length of the collected array (llen) */
 
+#if PIO_USE_MPISERIAL
+                    if ((mpierr = MPI_Type_hvector(nvars, 1, (MPI_Aint) iodesc->llen * tsize,
+                                                   iodesc->rtype[i], recvtypes + i)))
+                        return check_mpi(NULL, mpierr, __FILE__, __LINE__);
+#else
                     if ((mpierr = MPI_Type_create_hvector(nvars, 1, (MPI_Aint) iodesc->llen * tsize,
                                                           iodesc->rtype[i], recvtypes + i)))
                         return check_mpi(NULL, mpierr, __FILE__, __LINE__);
+#endif /* PIO_USE_MPISERIAL */
 
                     if (recvtypes[i] == PIO_DATATYPE_NULL)
                         return pio_err(NULL, NULL, PIO_EINVAL, __FILE__, __LINE__);
@@ -896,9 +902,15 @@ int rearrange_comp2io(iosystem_desc_t *ios, io_desc_t *iodesc, void *sbuf,
                 {
                     recvcounts[iodesc->rfrom[i]] = 1;
 
+#if PIO_USE_MPISERIAL
+                    if ((mpierr = MPI_Type_hvector(nvars, 1, (MPI_Aint)iodesc->llen * tsize,
+                                                   iodesc->rtype[i], recvtypes + iodesc->rfrom[i])))
+                        return check_mpi(NULL, mpierr, __FILE__, __LINE__);
+#else
                     if ((mpierr = MPI_Type_create_hvector(nvars, 1, (MPI_Aint)iodesc->llen * tsize,
                                                           iodesc->rtype[i], recvtypes + iodesc->rfrom[i])))
                         return check_mpi(NULL, mpierr, __FILE__, __LINE__);
+#endif /* PIO_USE_MPISERIAL */
 
                     if (recvtypes[iodesc->rfrom[i]] == PIO_DATATYPE_NULL)
                         return pio_err(NULL, NULL, PIO_EINVAL, __FILE__, __LINE__);
@@ -924,9 +936,15 @@ int rearrange_comp2io(iosystem_desc_t *ios, io_desc_t *iodesc, void *sbuf,
         if (scount[i] > 0 && sbuf != NULL)
         {
             sendcounts[io_comprank] = 1;
+#if PIO_USE_MPISERIAL
+            if ((mpierr = MPI_Type_hvector(nvars, 1, (MPI_Aint)iodesc->ndof * tsize,
+                                           iodesc->stype[i], sendtypes + io_comprank)))
+                return check_mpi(NULL, mpierr, __FILE__, __LINE__);
+#else
             if ((mpierr = MPI_Type_create_hvector(nvars, 1, (MPI_Aint)iodesc->ndof * tsize,
                                                   iodesc->stype[i], sendtypes + io_comprank)))
                 return check_mpi(NULL, mpierr, __FILE__, __LINE__);
+#endif /* PIO_USE_MPISERIAL */
 
             if (sendtypes[io_comprank] == PIO_DATATYPE_NULL)
                 return pio_err(NULL, NULL, PIO_EINVAL, __FILE__, __LINE__);

